### PR TITLE
Keep scroll position, listen message from iframe

### PR DESCRIPTION
**Pull request by MikeFuster**
https://user-images.githubusercontent.com/85583835/148832665-b957975f-5224-414e-9b02-9d555c5ef577.mov

https://user-images.githubusercontent.com/88199805/149802557-05e86d0c-374c-4fdc-9709-47a9fde259a3.mov

### CODE REVIEW

File: public/app/plugins/panel/logs/useNewData.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { useEffect, useState } from 'react';
+import { cloneDeep } from 'lodash';
+
+interface Props {
+  data: any;
+}
+
+const useNewData = ({ data }: Props) => {
```
Comment (fperaltalitebox) [inline public/app/plugins/panel/logs/useNewData.tsx:8]: @MikeFuster I'd change the name of this hook to something like usePopulatedData 💪

**Comment by bennbollay** [comment]
Odd things noticed:
1. Adding a log message when there's no data at all causes an exception:

```
useNewData.tsx:20 Uncaught TypeError: Cannot read properties of undefined (reading 'values')
    at addMessageToData (useNewData.tsx:20)
    at postMessage (useNewData.tsx:47)
```

2. Clicking on the dropdown arrow on an added log message uses labels from the wrong log entry:
image

**Comment by bennbollay** [comment]
Here is the current panel query line I'm using, with some filters removed from the front:
```
json | line_format "{{.msg}}"
```
We'll need to have a variety of metadata passed in, so use a JSON-format (either an object or a stringified object) as the payload, so it can be enriched with various values.

File: public/app/plugins/panel/logs/useNewData.tsx (hunk 1 above)
Comment (MikeFuster) [reply public/app/plugins/panel/logs/useNewData.tsx]: Done!

**Review by fperaltalitebox — APPROVED** [review]

**Comment by fperaltalitebox** [comment]
> Here is the current panel query line I'm using, with some filters removed from the front:
> 
> ```
> json | line_format "{{.msg}}"
> ```
> 
> We'll need to have a variety of metadata passed in, so use a JSON-format (either an object or a stringified object) as the payload, so it can be enriched with various values.

Fixed!

**Comment by fperaltalitebox** [comment]
> Odd things noticed:
> 
> 1. Adding a log message when there's no data at all causes an exception:
> 
> ```
> useNewData.tsx:20 Uncaught TypeError: Cannot read properties of undefined (reading 'values')
>     at addMessageToData (useNewData.tsx:20)
>     at postMessage (useNewData.tsx:47)
> ```
> 
> 2. Clicking on the dropdown arrow on an added log message uses labels from the wrong log entry:
>    image

Fixed!

**Review by bennbollay — APPROVED** [review]
Approved once the changes to conf/defaults.ini are reverted.

**Pull request merged**